File: BadgeGenerator.java
// BadgeGenerator.java
import java.awt.*;
import java.awt.geom.Ellipse2D;
import java.awt.image.BufferedImage;
import java.io.*;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.*;
import javax.imageio.ImageIO;

public class BadgeGenerator {
    private String name, title, company, photoPath;
    private int width, height, photoSize;
    private Color bgColor, textColor;
    private String photoShape;
    private Font font;

    public BadgeGenerator(String name, String title, String company, String photoPath,
                          int width, int height, Color bg, Color text, int photoSize,
                          String photoShape, Font font) {
        this.name = name;
        this.title = title;
        this.company = company;
        this.photoPath = photoPath;
        this.width = width;
        this.height = height;
        this.bgColor = bg;
        this.textColor = text;
        this.photoSize = photoSize;
        this.photoShape = photoShape;
        this.font = font;
    }

    public void generate(String output) throws Exception {
        // Загружаем фото
        BufferedImage photo = ImageIO.read(new File(photoPath));
        if (photo == null) throw new IOException("Не удалось загрузить фото");
        // Масштабируем
        Image scaledPhoto = photo.getScaledInstance(photoSize, photoSize, Image.SCALE_SMOOTH);
        BufferedImage photoImg = new BufferedImage(photoSize, photoSize, BufferedImage.TYPE_INT_ARGB);
        Graphics2D g2 = photoImg.createGraphics();
        g2.drawImage(scaledPhoto, 0, 0, null);
        if (photoShape.equals("circle")) {
            // Создаём круглую маску
            Ellipse2D circle = new Ellipse2D.Double(0, 0, photoSize, photoSize);
            g2.setClip(circle);
            g2.drawImage(scaledPhoto, 0, 0, null);
        }
        g2.dispose();

        // Создаём бейдж
        BufferedImage badge = new BufferedImage(width, height, BufferedImage.TYPE_INT_RGB);
        Graphics2D g = badge.createGraphics();
        g.setColor(bgColor);
        g.fillRect(0, 0, width, height);
        g.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);

        // Вставляем фото
        int photoX = 30;
        int photoY = (height - photoSize) / 2;
        g.drawImage(photoImg, photoX, photoY, null);

        // Текст
        int textX = photoX + photoSize + 30;
        int textY = (height - 90) / 2;
        g.setColor(textColor);
        Font nameFont = font != null ? font.deriveFont(Font.BOLD, 36) : new Font("Arial", Font.BOLD, 36);
        g.setFont(nameFont);
        g.drawString(name, textX, textY);
        if (title != null && !title.isEmpty()) {
            Font titleFont = font != null ? font.deriveFont(Font.PLAIN, 24) : new Font("Arial", Font.PLAIN, 24);
            g.setFont(titleFont);
            g.drawString(title, textX, textY + 40);
        }
        if (company != null && !company.isEmpty()) {
            Font compFont = font != null ? font.deriveFont(Font.PLAIN, 28) : new Font("Arial", Font.PLAIN, 28);
            g.setFont(compFont);
            g.drawString(company, textX, textY + 70);
        }
        g.dispose();

        // Сохраняем
        ImageIO.write(badge, "png", new File(output));
        System.out.println("Бейдж сохранён в " + output);
    }

    public static void main(String[] args) throws Exception {
        Map<String, String> params = new HashMap<>();
        for (int i = 0; i < args.length; i++) {
            if (args[i].startsWith("--")) {
                String key = args[i].substring(2);
                if (i + 1 < args.length && !args[i+1].startsWith("--")) {
                    params.put(key, args[++i]);
                } else {
                    params.put(key, "");
                }
            }
        }
        String name = params.getOrDefault("name", "");
        String title = params.getOrDefault("title", null);
        String company = params.getOrDefault("company", null);
        String photoPath = params.getOrDefault("photo", "");
        String output = params.getOrDefault("output", "badge.png");
        int width = Integer.parseInt(params.getOrDefault("width", "600"));
        int height = Integer.parseInt(params.getOrDefault("height", "400"));
        Color bg = Color.decode(params.getOrDefault("bg", "#FFFFFF"));
        Color textColor = Color.decode(params.getOrDefault("text-color", "#000000"));
        int photoSize = Integer.parseInt(params.getOrDefault("photo-size", "120"));
        String photoShape = params.getOrDefault("photo-shape", "circle");
        // Шрифт пока игнорируем (используем Arial)

        if (name.isEmpty() || photoPath.isEmpty()) {
            System.err.println("Ошибка: --name и --photo обязательны");
            System.exit(1);
        }
        BadgeGenerator gen = new BadgeGenerator(name, title, company, photoPath,
                width, height, bg, textColor, photoSize, photoShape, null);
        gen.generate(output);
    }
}

File: BadgeGenerator.kt
// BadgeGenerator.kt
import java.awt.*
import java.awt.geom.Ellipse2D
import java.awt.image.BufferedImage
import java.io.File
import javax.imageio.ImageIO
import kotlin.system.exitProcess

data class Options(
    val name: String,
    val title: String?,
    val company: String?,
    val photo: String,
    val output: String,
    val width: Int,
    val height: Int,
    val bg: Color,
    val textColor: Color,
    val photoSize: Int,
    val photoShape: String
)

fun parseArgs(args: Array<String>): Options {
    val map = mutableMapOf<String, String>()
    var i = 0
    while (i < args.size) {
        if (args[i].startsWith("--")) {
            val key = args[i].substring(2)
            if (i + 1 < args.size && !args[i+1].startsWith("--")) {
                map[key] = args[++i]
            } else {
                map[key] = ""
            }
        }
        i++
    }
    val name = map["name"] ?: ""
    val title = map["title"]
    val company = map["company"]
    val photo = map["photo"] ?: ""
    val output = map["output"] ?: "badge.png"
    val width = map["width"]?.toIntOrNull() ?: 600
    val height = map["height"]?.toIntOrNull() ?: 400
    val bg = Color.decode(map["bg"] ?: "#FFFFFF")
    val textColor = Color.decode(map["text-color"] ?: "#000000")
    val photoSize = map["photo-size"]?.toIntOrNull() ?: 120
    val photoShape = map["photo-shape"] ?: "circle"
    return Options(name, title, company, photo, output, width, height, bg, textColor, photoSize, photoShape)
}

fun generateBadge(opts: Options) {
    // Загружаем фото
    val photo = ImageIO.read(File(opts.photo)) ?: throw Exception("Не удалось загрузить фото")
    val scaledPhoto = photo.getScaledInstance(opts.photoSize, opts.photoSize, Image.SCALE_SMOOTH)
    val photoImg = BufferedImage(opts.photoSize, opts.photoSize, BufferedImage.TYPE_INT_ARGB)
    val g2 = photoImg.createGraphics()
    if (opts.photoShape == "circle") {
        val clip = Ellipse2D.Double(0.0, 0.0, opts.photoSize.toDouble(), opts.photoSize.toDouble())
        g2.clip = clip
    }
    g2.drawImage(scaledPhoto, 0, 0, null)
    g2.dispose()

    // Бейдж
    val badge = BufferedImage(opts.width, opts.height, BufferedImage.TYPE_INT_RGB)
    val g = badge.createGraphics()
    g.color = opts.bg
    g.fillRect(0, 0, opts.width, opts.height)
    g.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON)

    // Фото
    val photoX = 30
    val photoY = (opts.height - opts.photoSize) / 2
    g.drawImage(photoImg, photoX, photoY, null)

    // Текст
    val textX = photoX + opts.photoSize + 30
    val textY = (opts.height - 90) / 2
    g.color = opts.textColor
    val nameFont = Font("Arial", Font.BOLD, 36)
    g.font = nameFont
    g.drawString(opts.name, textX, textY)
    opts.title?.let {
        val titleFont = Font("Arial", Font.PLAIN, 24)
        g.font = titleFont
        g.drawString(it, textX, textY + 40)
    }
    opts.company?.let {
        val compFont = Font("Arial", Font.PLAIN, 28)
        g.font = compFont
        g.drawString(it, textX, textY + 70)
    }
    g.dispose()

    // Сохраняем
    ImageIO.write(badge, "png", File(opts.output))
    println("Бейдж сохранён в ${opts.output}")
}

fun main(args: Array<String>) {
    val opts = parseArgs(args)
    if (opts.name.isEmpty() || opts.photo.isEmpty()) {
        System.err.println("Ошибка: --name и --photo обязательны")
        exitProcess(1)
    }
    try {
        generateBadge(opts)
    } catch (e: Exception) {
        System.err.println("Ошибка: ${e.message}")
        exitProcess(1)
    }
}
